core/tests/multiplex.rs

```diff
@@ -170,22 +158,20 @@ fn connection_reused_for_dialing() {
         tokio_current_thread::block_on_all(future).unwrap();
     });
 
-    let transport = OnlyOnce::from(TcpConfig::new())
+    let transport = OnlyOnce::from(tx)
         .with_upgrade(multiplex::MplexConfig::new())
         .into_connection_reuse();
 
-    let listen_addr = rx.recv().unwrap();
-
     let future = transport
         .clone()
-        .dial(listen_addr.clone())
+        .dial("/memory".parse().unwrap())
         .unwrap_or_else(|_| panic!())
         .map(|server| Framed::<_, BytesMut>::new(server.0))
         .and_then(|server| server.send("hello world".into()))
         .and_then(|first_connec| {
             transport
                 .clone()
-                .dial(listen_addr.clone())
+                .dial("/memory".parse().unwrap())
                 .unwrap_or_else(|_| panic!())
```

Comment:
I guess you took it from here? Still unwrap already panics, why `panic!` in the `or_else` - that doesn't make any sense to me...?

Reply:
I'm not sure if that's what you mean, but the `unwrap()` applies to parsing `/memory` into a `Multiaddr`.